File: 55-jump-game/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if(nums.size() == 1) return true;
        if(nums[0] == 0) return false;
        int maxi = 0;
        for(int i=0;i<nums.size();i++){
            if(i > maxi) return false;
            maxi = max(i + nums[i],maxi);
        }
        return true;
    }
};

File: 55-jump-game/jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int temp=0;
        for(int i=0;i<nums.length;i++){
            temp=Math.max(temp,nums[i]+i);
            // System.out.println(temp);
            if(temp>=nums.length-1) return true;
            if(temp==i && nums[i]==0) return false;
        }
        return true;
    }
}